spring-cloud-dataflow-aggregate-task/src/main/java/org/springframework/cloud/dataflow/aggregate/task/impl/AggregateDataFlowTaskExecutionQueryDao.java

```diff
@@ -484,11 +485,12 @@ public TaskExecution mapRow(ResultSet rs, int rowNum) throws SQLException {
 			if (rs.wasNull()) {
 				parentExecutionId = null;
 			}
+			Timestamp endTimestamp = rs.getTimestamp("END_TIME");
 			return new TaskExecution(id,
 					getNullableExitCode(rs),
 					rs.getString("TASK_NAME"),
 					rs.getTimestamp("START_TIME").toLocalDateTime(),
-					rs.getTimestamp("END_TIME").toLocalDateTime(),
+					(endTimestamp != null) ? rs.getTimestamp("END_TIME").toLocalDateTime() : null,
```

Comment:
Why not use `endTimestamp` instead of `rs.getTimestamp("END_TIME")` as we already pulled that into `endTimestamp`?

Reply:
DOH!   Good catch